File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public class Gigasecond {
    private LocalDateTime _epoch;
    public Gigasecond(LocalDate moment) {
        _epoch = moment.atStartOfDay();
    }

    public Gigasecond(LocalDateTime moment) {
        _epoch = moment;
    }

    public LocalDateTime getDateTime() {
        return _epoch.plus((long)Math.pow(10,9), ChronoUnit.SECONDS);
    }
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.lang.Math.pow

class Gigasecond (val epoch: LocalDateTime) {
    constructor(dt: LocalDate) : this(dt.atStartOfDay()) {}
    val date: LocalDateTime =  epoch.plusSeconds(pow(10.0,9.0).toLong())
}
